Libraries/LibWeb/CSS/Parser/ArbitrarySubstitutionFunctions.cpp

```diff
@@ -145,29 +159,35 @@ static Vector<ComponentValue> replace_an_attr_function(DOM::AbstractElement& ele
 
     // 4. If syntax is null or the keyword raw-string, return a CSS <string> whose value is attr value.
     // NOTE: No parsing or modification of any kind is performed on the value.
-    if (!maybe_syntax.has_value() || maybe_syntax->equals_ignoring_ascii_case("raw-string"sv))
+    if (syntax.visit(
+            [](Empty) { return true; },
+            [](RawString) { return true; },
+            [](auto&) { return false; })) {
         return { Token::create_string(*attribute_value) };
-    auto syntax = maybe_syntax.release_value();
+    }
 
     // 5. Substitute arbitrary substitution functions in attr value, with «"attribute", attr name» as the substitution
-    //    context, then parse with a attr value, with syntax and el. If that succeeds, return the result; otherwise,
-    //    jump to the last step (labeled FAILURE).
+    //    context, then parse with a <syntax> attr value, with syntax and el. If that succeeds, return the result;
+    //    otherwise, jump to the last step (labeled FAILURE).
     auto parser = Parser::create(ParsingParams { element.element().document() }, attribute_value.value());
     auto unsubstituted_values = parser.parse_as_list_of_component_values();
-    auto substituted_values = substitute_arbitrary_substitution_functions(element, guarded_contexts, unsubstituted_values, SubstitutionContext { SubstitutionContext::DependencyType::Attribute, attribute_name.to_string() });
-
-    // FIXME: Parse using the syntax. For now we just handle `<attr-unit>` here.
-    TokenStream value_tokens { substituted_values };
-    value_tokens.discard_whitespace();
-    auto const& component_value = value_tokens.consume_a_token();
-    value_tokens.discard_whitespace();
-    if (value_tokens.has_next_token())
+    auto substituted_values = substitute_arbitrary_substitution_functions(element, guarded_contexts, unsubstituted_values,
+        SubstitutionContext { SubstitutionContext::DependencyType::Attribute, attribute_name.to_string() });
+
+    auto parsed_value = parse_with_a_syntax(ParsingParams { element.document() }, substituted_values, *syntax.get<NonnullOwnPtr<SyntaxNode>>(), element);
+    if (parsed_value->is_guaranteed_invalid())
         return failure();
 
-    if (component_value.is(Token::Type::Number) && is_dimension_unit)
-        return { Token::create_dimension(component_value.token().number_value(), syntax) };
+    if (unit_name.has_value()) {
+        // https://drafts.csswg.org/css-values-5/#ref-for-typedef-attr-type%E2%91%A0
+        // If given as an <attr-unit> value, the value is first parsed as if type(<number>) was specified, then the
+        // resulting numeric value is turned into a dimension with the corresponding unit, or a percentage if % was
+        // given. Values that fail to parse as a <number> trigger fallback.
+        VERIFY(parsed_value->is_number());
```

Comment:
`parsed_value` isn't guaranteed to be a number at this point. This causes a crash in https://wpt.live/css/css-values/crashtests/chrome-405422528-crash.html, where `parsed_value` is a calculated value.